### PR TITLE
fix: `appendOnPaste` and `mode=auto` not working together

### PR DESCRIPTION
**Pull request by juliusmarminge**

### CODE REVIEW

File: packages/react/src/components/dropzone.tsx [hunk 1]
```diff
@@ -149,13 +149,17 @@ export function UploadDropzone<
       if (document.activeElement !== rootRef.current) return;
 
       const pastedFiles = getFilesFromClipboardEvent(event);
-      if (!pastedFiles) return;
+      if (!pastedFiles?.length) return;
 
-      setFiles((prev) => [...prev, ...pastedFiles]);
+      let filesToUpload = pastedFiles;
+      setFiles((prev) => {
+        filesToUpload = [...prev, ...pastedFiles];
```
Comment (juliusmarminge) [inline packages/react/src/components/dropzone.tsx:156]: idk if this is cursed or good...

File: packages/react/src/components/dropzone.tsx [hunk 2]
```diff
@@ -149,13 +149,17 @@ export function UploadDropzone<
       if (document.activeElement !== rootRef.current) return;
 
       const pastedFiles = getFilesFromClipboardEvent(event);
-      if (!pastedFiles) return;
+      if (!pastedFiles?.length) return;
```
Comment (markflorkowski) [inline packages/react/src/components/dropzone.tsx:152]: Should we consider an [eslint plugin](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/HEAD/docs/rules/explicit-length-check.md) to enforce this?

File: packages/react/src/components/dropzone.tsx (hunk 1 above)
Comment (markflorkowski) [reply packages/react/src/components/dropzone.tsx:156]: How is this different from before....? 😅

**Review by markflorkowski — APPROVED** [review]

File: packages/react/src/components/dropzone.tsx (hunk 1 above)
Comment (juliusmarminge) [reply packages/react/src/components/dropzone.tsx:156]: State update cycle

Comment (markflorkowski) [reply packages/react/src/components/dropzone.tsx:156]: 🥲

**Pull request merged**